### PR TITLE
Print nil check log only once

### PR DESCRIPTION
**Pull request by yanweiguo**
It turns about this log is pretty annoying and generating hundreds of lines.

### CODE REVIEW

**Comment by yanweiguo** [comment]
> Should we keep at least `The metricsConfig has not been initialized yet` part?

Then it still throw hundreds of lines. What's your main pain point of this as you send a few PRs to avoid these logs in unit tests?

**Comment by vagababov** [comment]
Can we put log in sync.Once?
Then it'll log just once per binary

File: metrics/config.go [hunk 1]
```diff
@@ -155,14 +156,15 @@ func NewStackdriverClientConfigFromMap(config map[string]string) *StackdriverCli
...
 	if mc == nil {
-		log.Println(`The metricsConfig has not been initialized yet.
-
-If this is a Go unit test consuming metric.Record(...) or metric.RecordBatch(...) then
-it should add the following import:
-
-import (
-	_ "knative.dev/pkg/metrics/testing"
+		logOnce.Do(func() {
+			log.Println(`The metricsConfig has not been initialized yet.
```
Comment (mattmoor) [inline metrics/config.go:160]: Remove trailing whitespace:
```suggestion
			log.Println(`The metricsConfig has not been initialized yet.
```

File: metrics/config.go [hunk 2]
```diff
@@ -155,14 +156,15 @@ func NewStackdriverClientConfigFromMap(config map[string]string) *StackdriverCli
...
-		log.Println(`The metricsConfig has not been initialized yet.
-
-If this is a Go unit test consuming metric.Record(...) or metric.RecordBatch(...) then
-it should add the following import:
-
-import (
-	_ "knative.dev/pkg/metrics/testing"
+		logOnce.Do(func() {
+			log.Println(`The metricsConfig has not been initialized yet.	
+If this is a Go unit test consuming metric.Record(...) or metric.RecordBatch(...) then
```
Comment (mattmoor) [inline metrics/config.go:161]: Remove trailing whitespace:
```suggestion
If this is a Go unit test consuming metric.Record(...) or metric.RecordBatch(...) then
```

File: metrics/config.go [hunk 3]
```diff
@@ -155,14 +156,15 @@ func NewStackdriverClientConfigFromMap(config map[string]string) *StackdriverCli
...
-
-If this is a Go unit test consuming metric.Record(...) or metric.RecordBatch(...) then
-it should add the following import:
-
-import (
-	_ "knative.dev/pkg/metrics/testing"
+		logOnce.Do(func() {
+			log.Println(`The metricsConfig has not been initialized yet.	
+If this is a Go unit test consuming metric.Record(...) or metric.RecordBatch(...) then	
+it should add the following import:
```
Comment (mattmoor) [inline metrics/config.go:162]: Remove trailing whitespace:
```suggestion
it should add the following import:
```

File: metrics/config.go [hunk 4]
```diff
@@ -155,14 +156,15 @@ func NewStackdriverClientConfigFromMap(config map[string]string) *StackdriverCli
...
-If this is a Go unit test consuming metric.Record(...) or metric.RecordBatch(...) then
-it should add the following import:
-
-import (
-	_ "knative.dev/pkg/metrics/testing"
+		logOnce.Do(func() {
+			log.Println(`The metricsConfig has not been initialized yet.	
+If this is a Go unit test consuming metric.Record(...) or metric.RecordBatch(...) then	
+it should add the following import:	
+import (
```
Comment (mattmoor) [inline metrics/config.go:163]: Remove trailing whitespace:
```suggestion
import (
```

File: metrics/config.go [hunk 5]
```diff
@@ -155,14 +156,15 @@ func NewStackdriverClientConfigFromMap(config map[string]string) *StackdriverCli
...
-it should add the following import:
-
-import (
-	_ "knative.dev/pkg/metrics/testing"
+		logOnce.Do(func() {
+			log.Println(`The metricsConfig has not been initialized yet.	
+If this is a Go unit test consuming metric.Record(...) or metric.RecordBatch(...) then	
+it should add the following import:	
+import (	
+	_ "knative.dev/pkg/metrics/testing"
```
Comment (mattmoor) [inline metrics/config.go:164]: Remove trailing whitespace:
```suggestion
	_ "knative.dev/pkg/metrics/testing"
```

**Comment by yanweiguo** [comment]
> Can we put log in sync.Once?
> Then it'll log just once per binary

sg. Done.

**Pull request merged**